python/cudf/cudf/io/parquet.py

```diff
@@ -175,29 +174,47 @@ def read_parquet(
 ):
     """{docstring}"""
 
-    filepath_or_buffer, compression = ioutils.get_filepath_or_buffer(
-        filepath_or_buffer, None, **kwargs
-    )
-    if compression is not None:
-        raise ValueError("URL content-encoding decompression is not supported")
+    # Multiple sources are passed as a list. If a single source is passed,
+    # wrap it in a list for unified processing downstream.
+    if not isinstance(filepath_or_buffer, list):
+        filepath_or_buffer = [filepath_or_buffer]
+
+    # a list of row groups per source should be passed. If a single list is
+    # passed, wrap in into a list to make the list of lists that is expected
+    # for multiple sources
+    if row_group_list and not isinstance(row_group_list[0], list):
+        row_group_list = [row_group_list]
+
+    filepaths_or_buffers = []
+    for source in filepath_or_buffer:
+        tmp_source, compression = ioutils.get_filepath_or_buffer(
+            source, None, **kwargs
+        )
+        if compression is not None:
+            raise ValueError(
+                "URL content-encoding decompression is not supported"
+            )
+        filepaths_or_buffers.append(tmp_source)
 
     if engine == "cudf":
         return libparquet.read_parquet(
-            filepath_or_buffer,
+            filepaths_or_buffers,
             columns=columns,
-            row_group=row_group,
-            row_group_count=row_group_count,
+            row_group_lists=row_group_list,
             skip_rows=skip_rows,
             num_rows=num_rows,
             strings_to_categorical=strings_to_categorical,
             use_pandas_metadata=use_pandas_metadata,
         )
     else:
         warnings.warn("Using CPU via PyArrow to read Parquet dataset.")
-        pa_table = pq.read_pandas(
-            filepath_or_buffer, columns=columns, *args, **kwargs
-        )
-        return cudf.DataFrame.from_arrow(pa_table)
+        pa_tables = [
+            cudf.DataFrame.from_arrow(
+                pq.read_pandas(fob, columns=columns, *args, **kwargs)
+            )
+            for fob in filepaths_or_buffers
+        ]
+        return cudf.concat(pa_tables)
```

Comment:
Does arrow not support passing a vector of files down to it?

Reply:
Looks like it does not (at least I could not pass the list directly).
I found [a comment in an Arrow issue](https://issues.apache.org/jira/browse/ARROW-3424?focusedCommentId=17082957&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17082957) that may suggest that there is a way to do this, really not sure.

Reply:
Looks like we'd need to do `pq.ParquetDataset(filepaths_or_buffers).read()`: https://arrow.apache.org/docs/python/generated/pyarrow.parquet.ParquetDataset.html